micrometer-core/src/main/java/io/micrometer/core/instrument/step/StepTimer.java

```diff
@@ -30,82 +29,39 @@
 public class StepTimer extends AbstractTimer {
     private final StepLong count;
     private final StepLong total;
-    private final TimeWindowMax max;
+    private final StepLongMax max;
 
-    /**
-     * Create a new {@code StepTimer}.
-     *
-     * @param id                          ID
-     * @param clock                       clock
-     * @param distributionStatisticConfig distribution statistic configuration
-     * @param pauseDetector               pause detector
-     * @param baseTimeUnit                base time unit
-     * @deprecated Use {@link #StepTimer(io.micrometer.core.instrument.Meter.Id, Clock, DistributionStatisticConfig, PauseDetector, TimeUnit, long, boolean)}
-     */
-    @Deprecated
-    public StepTimer(Id id, Clock clock, DistributionStatisticConfig distributionStatisticConfig,
-                     PauseDetector pauseDetector, TimeUnit baseTimeUnit) {
-        this(id, clock, distributionStatisticConfig, pauseDetector, baseTimeUnit, false);
-    }
-
-    /**
-     * Create a new {@code StepTimer}.
-     *
-     * @param id                            ID
-     * @param clock                         clock
-     * @param distributionStatisticConfig   distribution statistic configuration
-     * @param pauseDetector                 pause detector
-     * @param baseTimeUnit                  base time unit
-     * @param supportsAggregablePercentiles whether it supports aggregable percentiles
-     * @deprecated Use {@link #StepTimer(io.micrometer.core.instrument.Meter.Id, Clock, DistributionStatisticConfig, PauseDetector, TimeUnit, long, boolean)}
-     */
-    @Deprecated
-    @SuppressWarnings("ConstantConditions")
-    public StepTimer(Id id, Clock clock, DistributionStatisticConfig distributionStatisticConfig,
-                     PauseDetector pauseDetector, TimeUnit baseTimeUnit, boolean supportsAggregablePercentiles) {
-        this(id, clock, distributionStatisticConfig, pauseDetector, baseTimeUnit, distributionStatisticConfig.getExpiry().toMillis(), supportsAggregablePercentiles);
-    }
-
-    /**
-     * Create a new {@code StepTimer}.
```

Comment:
We've lost the JavaDoc for the now only public constructor for a StepTimer with these changes. I can restore this post-merge.